src/ServiceMonitor/EtwListner.cpp

```diff
@@ -0,0 +1,271 @@
+    EVENT_TRACE_PROPERTIES iisLogSessionPropt;
+
+    //TRACEHANDLE handle;
+
+    TRACEHANDLE hTrace;
+
+    //
+    // stop the trace if the trace already started
+    //
+
+    closeSessionPropt.Wnode.BufferSize = 1000;
+    StopTrace(NULL, pStrSessionName, &closeSessionPropt);
+
+    if (uStatus != ERROR_SUCCESS)
+    {
+        _tprintf(L"UNABLE TO STOP OLD TRACE %i\n", uStatus);
+    }
+
+    ZeroMemory(&iisLogSessionPropt, sizeof(EVENT_TRACE_PROPERTIES));
+
+    iisLogSessionPropt.Wnode.BufferSize = sizeof(EVENT_TRACE_PROPERTIES) + wcslen(pStrSessionName) * WCHAR_BYTES + WCHAR_NULL_BYTES;
+
+    iisLogSessionPropt.Wnode.Flags = WNODE_FLAG_TRACED_GUID;
+    iisLogSessionPropt.Wnode.ClientContext = 1;
+    iisLogSessionPropt.LogFileMode = EVENT_TRACE_REAL_TIME_MODE | EVENT_TRACE_NO_PER_PROCESSOR_BUFFERING;
+    iisLogSessionPropt.FlushTimer = 1;
+    iisLogSessionPropt.MaximumFileSize = 1024;
+    iisLogSessionPropt.LoggerNameOffset = sizeof(EVENT_TRACE_PROPERTIES);
+    iisLogSessionPropt.LogFileNameOffset = 0;
+    iisLogSessionPropt.BufferSize = 1024;
+    iisLogSessionPropt.MinimumBuffers = 20;
+    iisLogSessionPropt.MaximumBuffers = 30;
+
+    EVENT_TRACE_LOGFILE eventTraceLog = { 0 };
+
+    uStatus = StartTrace((PTRACEHANDLE)&hTrace, pStrSessionName, &iisLogSessionPropt);
+    if (uStatus != ERROR_SUCCESS)
+    {
+        _tprintf(L"START TRACE FAIL %i\n", uStatus);
+        goto finish;
+    }
+
+    uStatus = EnableTraceEx2(hTrace,
+        (LPCGUID)pTraceGUID,
+        EVENT_CONTROL_CODE_ENABLE_PROVIDER,
+        TRACE_LEVEL_INFORMATION,
+        0,
+        0,
+        0,
+        NULL
+    );
+
+    if (uStatus != ERROR_SUCCESS)
+    {
+        _tprintf(L"ENABLE TRACE FAIL \n");
+        goto finish;
+    }
+
+    ZeroMemory(&eventTraceLog, sizeof(EVENT_TRACE_LOGFILE));
+
+    eventTraceLog.LogFileName = NULL;
+
+    eventTraceLog.LoggerName = pStrSessionName;
+    eventTraceLog.CurrentTime = 0;
+    eventTraceLog.BuffersRead = 0;
+
+    eventTraceLog.ProcessTraceMode = PROCESS_TRACE_MODE_EVENT_RECORD | PROCESS_TRACE_MODE_REAL_TIME;
+
+    eventTraceLog.BufferCallback = NULL;
+    eventTraceLog.BufferSize = 0;
+    eventTraceLog.Filled = 0;
+    eventTraceLog.EventsLost = 0;
+
+    eventTraceLog.EventRecordCallback = IISLoggingEvtRecCallback;
+    eventTraceLog.Context = NULL;
+
+    //
+    // reuse the hTrace handle variable
+    //
+
+    hTrace = OpenTrace(&eventTraceLog);
+    if (hTrace == INVALID_PROCESSTRACE_HANDLE)
+    {
+        uStatus = GetLastError();
+        _tprintf(L"OPEN TRACE FAIL %i\n", uStatus);
+        goto finish;
+    }
+
+    pThreadHandle = CreateThread(NULL,
+        0x2000000,
+        ProcessEtwThreadProc,
+        (LPVOID)&hTrace,
+        0,
+        &dwThreadId);
+
+    if (pThreadHandle == NULL)
+    {
+        uStatus = GetLastError();
+        _tprintf(L"START THREAD FAIL %i\n", uStatus);
+        goto finish;
+    }
+    std::wcout << "BYE" << std::endl;

```

Comment:
This trace should be more informative. _BYE_ can easily get lost within the console. If this trace is needed how about _Shutting down ETW Listener..._

Reply:
This is debug info. Will delete this.